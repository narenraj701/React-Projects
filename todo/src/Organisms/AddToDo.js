import React,{useState} from 'react';
import axios from 'axios';
import {makeStyles} from '@material-ui/core/styles'
import {Button, Typography} from '@material-ui/core';

const useStyles=makeStyles((theme) => ({
    root: {
      color:'white',
      backgroundColor:'green',
      margin:'5px',
      '&:hover':{
        opacity:0.8,
        color:'white',
        backgroundColor:'green'
      }
    },
    position:{
        textAlign:'center'
    },
    inputstyle:{
        width:'50%',
        height:'30px'
    },
}));

const AddToDo=(props)=>{
    const classes=useStyles();
    const [message,setMessage]=useState("");
    const [date,setDate]=useState("");
    const postHandler=()=>{
        const data={
            message:message,
            date:date,
            completed:false
        }
        axios.post("/api/messages",data)
        .then(response=>{
            window.location.reload(false);
        }).catch(error=>console.log(error));
    }
    
    return <div className={classes.position}>
        <Typography variant='h3'>Add A To Do</Typography>
        <input type='text' placeholder='enter a to do task' className={classes.inputstyle}  onChange={(event)=>setMessage(event.target.value)} /><br /><br />
        <input type='text' placeholder='enter due date'  className={classes.inputstyle} onChange={(event)=>setDate(event.target.value)}/><br /><br />
        <Button  className={classes.root} onClick={postHandler} message={message} date={date}>Submit</Button>
    </div> 
}

export default AddToDo; 