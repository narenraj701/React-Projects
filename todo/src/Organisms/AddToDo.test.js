
import React from "react";
import { create } from "react-test-renderer";
import { render,fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import AddToDo from './AddToDo';


test("Match the snapshot - Due message",() => {
    const input=create(<AddToDo />);
    expect(input.toJSON()).toMatchSnapshot();
})
test("onchange of task message", () => {
    const { getByPlaceholderText } = render(<AddToDo />);
    const messageInput = getByPlaceholderText("enter a to do task");
    fireEvent.change(messageInput, { target: { value: "complete work" } });
    expect(messageInput).toHaveValue("complete work");
  });

  test("onchange of due message", () => {
    const { getByPlaceholderText } = render(<AddToDo />);
    const dueInput = getByPlaceholderText("enter a to do task");
    fireEvent.change(dueInput, { target: { value: "17th July" } });
    expect(dueInput).toHaveValue("17th July");
  })
  test("Submit Button is present",()=>{ 
    const { getByRole } = render(<AddToDo />);
    const button=getByRole('button');
    fireEvent.click(button);
    expect(button).toBeInTheDocument();
  })