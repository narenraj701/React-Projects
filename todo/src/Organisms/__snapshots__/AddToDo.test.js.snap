// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot - Due message 1`] = `
<div
  className="makeStyles-position-2"
>
  <h3
    className="MuiTypography-root MuiTypography-h3"
  >
    Add A To Do
  </h3>
  <input
    className="makeStyles-inputstyle-3"
    onChange={[Function]}
    placeholder="enter a to do task"
    required={true}
    type="text"
  />
  <br />
  <br />
  <input
    className="makeStyles-inputstyle-3"
    onChange={[Function]}
    placeholder="enter due date"
    required={true}
    type="text"
  />
  <br />
  <br />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-1"
    date=""
    disabled={false}
    message=""
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Submit
    </span>
  </button>
</div>
`;
